### ISSUE TITLE
The axis name set by `torch.export.Dim` is not in ExportedProgram

### ISSUE THREAD

**Issue by titaiwangms** [pytorch/pytorch#144273]
### 🐛 Describe the bug

```python
import torch

class Model(torch.nn.Module):
    def forward(self, x, y):
        return x + y

dim = torch.export.Dim("batch", min=1, max=6)
ep = torch.export.export(
    Model(),
    (torch.randn(2, 3), torch.randn(2, 3)),
    dynamic_shapes=[{0: dim}, {0: dim}],
)
```
```
ExportedProgram:
    class GraphModule(torch.nn.Module):
        def forward(self, x: "f32[s0, 3]", y: "f32[s0, 3]"):
             # File: /home/titaiwang/pytorch/test_export.py:5 in forward, code: return x + y
            add: "f32[s0, 3]" = torch.ops.aten.add.Tensor(x, y);  x = y = None
            return (add,)
            
Graph signature: ExportGraphSignature(input_specs=[InputSpec(kind=<InputKind.USER_INPUT: 1>, arg=TensorArgument(name='x'), target=None, persistent=None), InputSpec(kind=<InputKind.USER_INPUT: 1>, arg=TensorArgument(name='y'), target=None, persistent=None)], output_specs=[OutputSpec(kind=<OutputKind.USER_OUTPUT: 1>, arg=TensorArgument(name='add'), target=None)])
Range constraints: {s0: VR[1, 6]}
```

The graph module and range constraints still use s0 to represent the dynamic axis.

### Versions

Nightly

Versions of relevant libraries:
[pip3] flake8==6.1.0
[pip3] flake8-bugbear==23.3.23
[pip3] flake8-coding==1.3.3
[pip3] flake8-comprehensions==3.15.0
[pip3] flake8-executable==2.1.3
[pip3] flake8-logging-format==0.9.0
[pip3] flake8-pyi==23.3.1
[pip3] flake8-simplify==0.19.3
[pip3] model-explorer-onnx==0.3.0
[pip3] mypy==1.13.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] onnx==1.17.0
[pip3] onnxruntime==1.20.1
[pip3] onnxscript==0.1.0.dev20241216
[pip3] optree==0.13.0
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] pytorch-triton==3.1.0+cf34004b8a
[pip3] torch==2.6.0a0+gitb5b1e94
[pip3] torchaudio==2.5.0a0+b4a286a
[pip3] torchmetrics==1.5.2
[pip3] torchvision==0.20.0a0+945bdad
[pip3] triton==3.2.0
[conda] mkl                       2023.1.0         h213fc3f_46344  
[conda] mkl-include               2023.1.0         h06a4308_46344  
[conda] mkl-service               2.4.0           py311h5eee18b_1  
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] optree                    0.13.0                   pypi_0    pypi
[conda] pytorch-sphinx-theme      0.0.24                    dev_0    <develop>
[conda] pytorch-triton            3.1.0+cf34004b8a          pypi_0    pypi
[conda] torch                     2.6.0a0+gitb5b1e94           dev_0    <develop>
[conda] torchaudio                2.5.0a0+b4a286a           dev_0    <develop>
[conda] torchfix                  0.4.0                    pypi_0    pypi
[conda] torchmetrics              1.5.2                    pypi_0    pypi
[conda] torchvision               0.20.0a0+945bdad           dev_0    <develop>
[conda] triton                    3.2.0                    pypi_0    pypi

cc @chauhang @penguinwu @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

**Comment by zhxchen17** [comment]
cc @pianpwk

**Comment by avikchaudhuri** [comment]
I'm not sure we can get away with using things like `s0` because the prefixes are significant for the compiler. Although I'm not entirely sure about this. 

However we could at least save the user-facing names as metadata.

**Comment by zhxchen17** [comment]
sorry @justinchuby @titaiwangms I think we had some idea to support but I'm out of bandwidth to support for now. Will leave it up for grab.
